src/python/pants/backend/native/rules/targets.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Dict, Iterable, Optional, Tuple, Union
+
+from pants.backend.native.subsystems.native_build_step import ToolchainVariant
+from pants.backend.native.targets.external_native_library import ConanRequirement
+from pants.backend.native.targets.native_artifact import NativeArtifact
+from pants.build_graph.address import Address
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    Dependencies,
+    InvalidFieldException,
+    InvalidFieldTypeException,
+    PrimitiveField,
+    Sources,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+from pants.util.collections import ensure_list, ensure_str_list
+from pants.util.frozendict import FrozenDict
+
+
+class CSources(Sources):
+    default = ("*.h", "*.c")
+
+
+class CppSources(Sources):
+    default = ("*.h", "*.hpp", "*.cpp")
+
+
+class CtypesNativeLibrary(PrimitiveField):
+    alias = "ctypes_native_library"
+    value: NativeArtifact
+    required = True
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[NativeArtifact], *, address: Address
+    ) -> NativeArtifact:
+        value = super().compute_value(raw_value, address=address)
+        if not isinstance(value, NativeArtifact):
+            raise InvalidFieldTypeException(
+                address, cls.alias, value, expected_type="a `native_artifact` object"
+            )
+        return value
+
+
+class NativeFatalWarnings(BoolField):
+    alias = "fatal_warnings"
+    default = False
+
+
+class NativeStrictDeps(BoolField):
```

Comment:
I wonder how this is different from the JVM's `strict_deps` and whether we should have a common field for it.

Reply:
I believe it's the same, modulo the docstring saying `native code`. We could indeed import from `backend/jvm` once those bindings land.

But, because we aren't actually consuming these targets in any rules yet, and are solely writing them so that we can remove `TargetAdaptor`, I'm going to stick with this and leave it to a future person to decide. I can add a comment.

Reply:
A comment would be great :) Thanks!